Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -545,6 +575,120 @@ namespace AzToolsFramework
+            {
+                newAttribute->m_alignment = alignment.value();
+            }
+            if (sharePrior)
+            {
+                AZ_Assert(priorColumnIndex != -1, "Tried to share column with an out of bounds index!");
+                if (priorColumnIndex != -1)
+                {
+                    m_columnLayout->AddSharePriorColumn(priorColumnIndex, domIndex);
+                    newAttribute->m_sharePriorColumn = true;
+                }
+            }
+            if (minimumWidth)
+            {
+                newAttribute->m_minimumWidth = true;
+            }
+            m_domOrderToAttributeInfo[domIndex] = newAttribute;
+        }
+        // An attribute entry already exists at this index, either update it or remove it
+        else if (foundEntry != m_domOrderToAttributeInfo.end())
+        {
+            // If none of the extracted attributes have value, delete the attribute entry at this index
+            if (!alignment.has_value() && !sharePrior && !minimumWidth)
+            {
+                m_columnLayout->RemoveSharePriorColumn(domIndex);
+                m_domOrderToAttributeInfo.erase(foundEntry);
+            }
+            // At least one attribute has been extracted, check if we need to update the existing attribute entry
+            else
+            {
+                // Check Alignment
+                if (alignment.has_value() && alignment.value() != foundEntry->second->m_alignment)
+                {
+                    foundEntry->second->m_alignment = alignment.value();
+                }
+
+                // Check SharePrior
+                if (sharePrior && !foundEntry->second->m_sharePriorColumn)
+                {
+                    AZ_Assert(priorColumnIndex != -1, "Tried to share column with an out of bounds index!")
+                    if (priorColumnIndex != -1)
+                    {
+                        m_columnLayout->AddSharePriorColumn(priorColumnIndex, domIndex);
+                        foundEntry->second->m_sharePriorColumn = true;
+                    }
+                }
+                else if (!sharePrior && foundEntry->second->m_sharePriorColumn)
+                {
+                    m_columnLayout->RemoveSharePriorColumn(domIndex);
+                    foundEntry->second->m_sharePriorColumn = false;
+                }
+
+                //Check MinimumWidth
+                if (minimumWidth != foundEntry->second->m_minimumWidth)
+                {
+                    foundEntry->second->m_minimumWidth = minimumWidth;
+                }
+            }
+        }
+    }
+
+    AzToolsFramework::DPERowWidget::AttributeInfo* DPERowWidget::GetAttributes(size_t domIndex)
+    {
+        auto foundEntry = m_domOrderToAttributeInfo.find(domIndex);
+        if (foundEntry != m_domOrderToAttributeInfo.end())
+        {
+            return foundEntry->second;
+        }
+        else
+        {
+            return new AttributeInfo;
+        }
+    }
```

Comment:
I think ideally I would want to return the attributes only if they exist, since this creates a default AttributeInfo every time, so maybe have a separate method that returns true if an attribute exists at this domIndex?

Reply:
I probably would just return null here, so the caller knows it doesn't exist yet, and then let them make a new one themselves, if that's reasonable. It's not like this else case is doing anything special...

Reply:
Good call, will do that